### DISCUSSION TITLE
Ping gateway

### DISCUSSION THREAD

**Discussion by nmcilree** [AmedeeBulle/pyrak811#25] [General]
Hi - I am trying to connect to TTN as per your README but keep getting RAK811 response error -1: Invalid argument.

Is there anyway to check if my device can see/ping a nearby gateway - just so I can rule this out as the source of the issue.

**Comment by AmedeeBulle** [comment]
Hi,

There is unfortunately no way to check for a gateway in range -- the best you could do is to look at the gateways placement on the TTN community pages.

Now I don't think you would get `error -1: Invalid argument` in that case. Maybe posting a more complete log of what you do would help to diagnose.

If you use the command line tool, use the `--verbose` flag (together with `--debug` if you have a v3 module)

**Comment by nmcilree** [comment]
Hi - I was running through the V2 example

```
`#!/usr/bin/env python3
# V2.0.x firmware
from rak811.rak811 import Mode, Rak811

lora = Rak811()
lora.hard_reset()
lora.mode = Mode.LoRaWan
lora.band = 'EU868'
lora.set_config(app_eui='70B3D5xxxxxxxxxx',
                app_key='xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx')
lora.join_otaa()
lora.dr = 5
lora.send('Hello world')
lora.close()`
```

The error appears on lora.join_otaa() - from both this script and command line.  I can get the dev_eui and eversion from the rak811 interface so I know that it is speaking to the PI.

I'm trying to work out whether the problem is with my TTN credentials or whether the rak811 is unable to connect to a LoraWan network for some other reason.

**Comment by AmedeeBulle** [comment]
If it was a connection issue, you would get a join failed or a timeout.

Invalid argument on the join command is really strange as the join doesn't take any argument other than the join mode (otaa). I suspect that a saved parameter is not compatible with OTAA or the module is in a wrong mode...

You might want to trace the dialog with the module; adding the following at the top of the script:

```python
import logging
logging.basicConfig(level=logging.DEBUG)
```

Can you also provide the exact version of your firmware?
```
rak811 version
```

**Reply by nmcilree** [reply]
Hi thanks -  here is the output from the debugger

DEBUG:rak811.serial:Serial initialized
DEBUG:rak811.serial:Ignoring untagged: >Welcome to RAK811<
DEBUG:rak811.serial:Ignoring untagged: ><
DEBUG:rak811.serial:Ignoring untagged: >Selected LoraWAN 1.0.2 Region: EU868 <
DEBUG:rak811.serial:Ignoring untagged: ><
DEBUG:rak811.serial:Sending: >at+mode=0\r\n<
DEBUG:rak811.serial:Ignoring untagged: ><
DEBUG:rak811.serial:Ignoring untagged: >Selected LoraWAN 1.0.2 Region: EU868 <
DEBUG:rak811.serial:Ignoring untagged: ><
DEBUG:rak811.serial:Received: >OK<
DEBUG:rak811.serial:Sending: >at+band=EU868\r\n<
DEBUG:rak811.serial:Received: >OK<
DEBUG:rak811.serial:Sending: >at+set_config=app_eui:0000000000000000&app_key:<-----------APP-KEY--------->\r\n<
DEBUG:rak811.serial:Received: >OK<
DEBUG:rak811.serial:Sending: >at+join=otaa\r\n<
DEBUG:rak811.serial:Received: >ERROR-1<

**Reply by nmcilree** [reply]
rak811 version returns this

---------  version  -------------
OK2.0.3.0
2.0.3.0

**Comment by AmedeeBulle** [comment]
Just to be sure, did you redacted the `app_eui` in the debug output or are you really setting it to all zeroes?

I'll reinstall a V2 module over the week-end to try to reproduce...

**Reply by nmcilree** [reply]
Thank you - I appreciate you help with this.

It is set to all zeros - I can't find the datasheet for my phat unit and my understanding of the TTN instructions is that all zeros would still work.

BTW it looks like there is an extra stage in setting up end devices in TTN not that differs from all the online instructions - you have to pick band, region and mac.

**Comment by AmedeeBulle** [comment]
Ah! This is the issue -- the module won't work without a proper `app_eui`

You can retrieve the `app_eui` from your device screen. If you left it empty at device creation, the console has populate it.
Simply copy/paste that value.

For the device eui of the module you can get it with `rak811 get-config dev_eui`

<img width="583" alt="rak" src="https://user-images.githubusercontent.com/4170863/141506133-18d949e1-60ec-4697-9595-75e1fffd640f.png">

Edit: actually this has been changed in TTN V3 -- I have my app eui as I migrated these from legacy TTN.

Need to investigate that.

**Comment by AmedeeBulle** [comment]
From [this post](https://github.com/mcci-catena/arduino-lmic/issues/690#issuecomment-792336037), using `0x000000001` as AppEUI should work...

**Comment by nmcilree** [comment]
HI - I followed the suggestion in the post but it did not work. However, thanks to you digging out the post explaining you can't use all zeros (despite instructions in TTN setup) I tried with 0100000000000009 and it's now connecting.

Thanks for helping with this - I was totally banging my head against a wall.

**Comment by AmedeeBulle** [comment]
It is indeed confusing.

The AppEUI (aka JoinEUI) should be provided by the device manufacturer (for commercial devices) or can be set to 0x00  for development purpose.

Unfortunately, the RAK module doesn't support AppEUI set to 0x00, and there is nothing I can do about this -- this is a RAK issue...

Using a random AppEUI will work most of the time, although it is not entirely legit as it might conflict with an existing officially assigned EUI.

For more confusion you can read [this post](https://forum.rakwireless.com/t/rak-7205-5205-setup-and-ttn-v3/4366/8) (and its reply) 🤷

**Reply by nmcilree** [reply]
Yeah - with the notable exception of your library I have found it difficult to find reliable/consistent information both from TTN ( out of date tutorials ) and the supply of my Pi RAK811 ( with regards data sheets )

Thanks once again.
